engine/src/flutter/shell/platform/android/surface_texture_external_texture_gl_impeller.cc

```diff
@@ -55,6 +55,7 @@ void SurfaceTextureExternalTextureGLImpeller::ProcessFrame(
 
 void SurfaceTextureExternalTextureGLImpeller::Detach() {
   SurfaceTextureExternalTexture::Detach();
+  texture_->Leak();
```

Comment:
Let's leave a comment mentioning the docstring saying that the detach will collect the handle.